go/internal/feast/onlinestore/cassandraonlinestore.go

```diff
@@ -0,0 +1,299 @@
+		tableName,
+		strings.Join(keyPlaceholders, ","),
+		strings.Join(quotedFeatureNames, ","),
+	)
+}
+
+func (c *CassandraOnlineStore) buildCassandraEntityKeys(entityKeys []*types.EntityKey) ([]interface{}, map[string]int, error) {
+	cassandraKeys := make([]interface{}, len(entityKeys))
+	cassandraKeyToEntityIndex := make(map[string]int)
+	for i := 0; i < len(entityKeys); i++ {
+		var key, err = utils.SerializeEntityKey(entityKeys[i], c.config.EntityKeySerializationVersion)
+		if err != nil {
+			return nil, nil, err
+		}
+		// encoding to hex
+		encodedKey := hex.EncodeToString(*key)
+		cassandraKeys[i] = encodedKey
+		cassandraKeyToEntityIndex[encodedKey] = i
+	}
+	return cassandraKeys, cassandraKeyToEntityIndex, nil
+}
+func (c *CassandraOnlineStore) OnlineRead(ctx context.Context, entityKeys []*types.EntityKey, featureViewNames []string, featureNames []string) ([][]FeatureData, error) {
+	uniqueNames := make(map[string]int32)
+	for _, fvName := range featureViewNames {
+		uniqueNames[fvName] = 0
+	}
+	if len(uniqueNames) != 1 {
+		return nil, fmt.Errorf("rejecting OnlineRead as more than 1 feature view was tried to be read at once")
+	}
+
+	serializedEntityKeys, serializedEntityKeyToIndex, err := c.buildCassandraEntityKeys(entityKeys)
+
+	if err != nil {
+		return nil, fmt.Errorf("error when serializing entity keys for Cassandra")
+	}
+	results := make([][]FeatureData, len(entityKeys))
+	for i := range results {
+		results[i] = make([]FeatureData, len(featureNames))
+	}
+
+	featureNamesToIdx := make(map[string]int)
+	for idx, name := range featureNames {
+		featureNamesToIdx[name] = idx
+	}
+
+	featureViewName := featureViewNames[0]
+
+	// Prepare the query
+	tableName := c.getFqTableName(featureViewName)
+	cqlStatement := c.getCQLStatement(tableName, featureNames, len(serializedEntityKeys))
+	// Bundle the entity keys in one statement (gocql handles this as concurrent queries)
+	scanner := c.session.Query(cqlStatement, serializedEntityKeys...).Iter().Scanner()
+
+	// Process the results
+	var entityKey string
+	var featureName string
+	var eventTs time.Time
+	var valueStr []byte
+	var deserializedValue types.Value
+	for scanner.Next() {
+		err := scanner.Scan(&entityKey, &featureName, &eventTs, &valueStr)
+		if err != nil {
+			return nil, errors.New("could not read row in query for (entity key, feature name, value, event ts)")
+		}
+		if err := proto.Unmarshal(valueStr, &deserializedValue); err != nil {
+			return nil, errors.New("error converting parsed Cassandra Value to types.Value")
+		}
+
+		var featureValues FeatureData
+		if deserializedValue.Val != nil {
+			// Convert the value to a FeatureData struct
+			featureValues = FeatureData{
+				Reference: serving.FeatureReferenceV2{
+					FeatureViewName: featureViewName,
+					FeatureName:     featureName,
+				},
+				Timestamp: timestamppb.Timestamp{Seconds: eventTs.Unix(), Nanos: int32(eventTs.Nanosecond())},
```

Comment:
timestamp value coming as 0 hours, 0 mins and 0 seconds in the example shared. Please cross check it.